api/v1alpha3/types.go

```diff
@@ -426,3 +426,12 @@ func (n *NetworkSpec) GetNodeSubnet() *SubnetSpec {
 	}
 	return nil
 }
+
+// SecurityProfile specifies the Security profile settings for a
+// virtual machine or virtual machine scale set.
+type SecurityProfile struct {
+	// This field indicates whether Host Encryption should be enabled
+	// or disabled for a virtual machine or virtual machine scale
+	// set. Default is disabled.
+	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
```

Comment:
Do we expect this field to be changed after initial cluster creation or is it immutable? If it's the later, it would be good to add webhook validation to make sure it doesn't get updated

Reply:
The [doc](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disk-encryption#restrictions-1) says:
> The encryption can be enabled on existing virtual machine scale set. However, only new VMs created after enabling the encryption are automatically encrypted.

Not sure how interpret it. Maybe prohibit modification for `AzureMachine`, but allow for `AzureMachineTemplate`?

Reply:
Ok that means the VMSS model can be updated so that's fine for AzureMachinePools 👍  machines are immutable already so you can update a machine deployment but an individual machine update won't trigger a VM update.